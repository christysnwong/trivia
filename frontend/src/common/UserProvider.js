import React, {useEffect, useState} from "react";
import UserContext from "./UserContext";

const demoUser = {
  id: 1,
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "test@test.com",
  stats: {
    level: 14,
    title: "Pro",
    "quizzes completed": 34,
    points: 3440,
    remainingPts: 60,
    levelPts: 300,
  },
};

const UserProvider = ({
  children,
  currUser = demoUser
}) => {
  
  const [currentUser, setCurrUser] = useState();

  return (
    // <UserContext.Provider value={{ currUser }}>
    <UserContext.Provider value={{ currUser, setCurrUser }}>
      {children}
    </UserContext.Provider>
  );
};

export default UserProvider;
