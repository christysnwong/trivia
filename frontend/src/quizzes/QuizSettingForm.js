import { useState } from "react";

const QuizSettingForm = ({ getQuizzes }) => {
  // set initial values for the quiz form
  const initValues = {
    category: "",
    difficulty: "",
  };

  const [formData, setFormData] = useState(initValues);

  // handle changes in the form
  const handleChange = (evt) => {
    const { name, value } = evt.target;

    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };

  // handle form submission and reset form
  const handleSubmit = async (evt) => {
    evt.preventDefault();

    await getQuizzes(formData);
    setFormData(initValues);
  };

  return (
    <div>
      <h3 className="mb-3">Quiz Form</h3>

      <div className="card">
        <div className="card-body">
          <form onSubmit={handleSubmit}>
            <div className="mb-3">
              <label htmlFor="category" className="form-label">
                Category
              </label>
              <select
                id="category"
                name="category"
                className="form-select"
                value={formData.category}
                onChange={handleChange}
                required
              >
                <option value="" disabled>
                  Choose here
                </option>
                <option value="9">General Knowledge</option>
                <option value="10">Entertainment: Books</option>
                <option value="32">
                  Entertainment: Cartoon and Animations
                </option>
                <option value="11">Entertainment: Film</option>
                <option value="31">
                  Entertainment: Japanese Anime and Manga
                </option>
                <option value="22">Geography</option>
                <option value="23">History</option>
                <option value="17">Science and Nature</option>
                <option value="18">Science: Computers</option>
                <option value="19">Science: Mathematics</option>
              </select>
            </div>
            <div className="mb-3">
              <label htmlFor="difficulty" className="form-label">
                Difficulty
              </label>
              <select
                id="difficulty"
                name="difficulty"
                className="form-select"
                value={formData.difficulty}
                onChange={handleChange}
                required
              >
                <option value="" disabled>
                  Choose here
                </option>
                <option value="easy">Easy</option>
                <option value="medium">Medium</option>
                <option value="hard">Hard</option>
              </select>
            </div>
            <button className="btn btn-primary float-end mt-4" type="submit">
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  );
};

export default QuizSettingForm;
