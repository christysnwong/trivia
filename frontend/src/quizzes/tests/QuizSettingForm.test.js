import React from "react";
import { render } from "@testing-library/react";
import QuizSettingForm from "../QuizSettingForm";
import { MemoryRouter } from "react-router-dom";
import UserProvider from "../../common/UserProvider";

it("renders without crashing", function () {
  render(
    <UserProvider>
      <QuizSettingForm />
    </UserProvider>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <QuizSettingForm />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
