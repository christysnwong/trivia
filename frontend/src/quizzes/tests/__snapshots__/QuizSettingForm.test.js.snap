// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3
      class="mb-3"
    >
      Quiz Form
    </h3>
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <form>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="category"
            >
              Category
            </label>
            <select
              class="form-select"
              id="category"
              name="category"
              required=""
            >
              <option
                disabled=""
                value=""
              >
                Choose here
              </option>
              <option
                value="9"
              >
                General Knowledge
              </option>
              <option
                value="10"
              >
                Entertainment: Books
              </option>
              <option
                value="32"
              >
                Entertainment: Cartoon and Animations
              </option>
              <option
                value="11"
              >
                Entertainment: Film
              </option>
              <option
                value="31"
              >
                Entertainment: Japanese Anime and Manga
              </option>
              <option
                value="22"
              >
                Geography
              </option>
              <option
                value="23"
              >
                History
              </option>
              <option
                value="17"
              >
                Science and Nature
              </option>
              <option
                value="18"
              >
                Science: Computers
              </option>
              <option
                value="19"
              >
                Science: Mathematics
              </option>
            </select>
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="difficulty"
            >
              Difficulty
            </label>
            <select
              class="form-select"
              id="difficulty"
              name="difficulty"
              required=""
            >
              <option
                disabled=""
                value=""
              >
                Choose here
              </option>
              <option
                value="easy"
              >
                Easy
              </option>
              <option
                value="medium"
              >
                Medium
              </option>
              <option
                value="hard"
              >
                Hard
              </option>
            </select>
          </div>
          <button
            class="btn btn-primary float-end mt-4"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
